### PR TITLE
[YUNIKORN-1826] Convert shim to use scoped logging

### PR DESCRIPTION
**Pull request by FrankYang0529**
### What is this PR for?

Convert all uses of log.Logger() to log.Log(name). This allows for fine-grained control over logging for each subsystem.

### What type of PR is it?
* [ ] - Bug Fix
* [X] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?

https://issues.apache.org/jira/browse/YUNIKORN-1826

### How should this be tested?

No functionality is changed, only the logging method.

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/common/si_helper.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ func CreateTagsForTask(pod *v1.Pod) map[string]string {
 			if value.Kind == constants.DaemonSetType {
 				if pod.Spec.Affinity == nil || pod.Spec.Affinity.NodeAffinity == nil ||
 					pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution == nil {
-					log.Logger().Debug("DaemonSet pod's Affinity, NodeAffinity, RequiredDuringSchedulingIgnoredDuringExecution might empty")
+					log.Log(log.ShimContext).Debug("DaemonSet pod's Affinity, NodeAffinity, RequiredDuringSchedulingIgnoredDuringExecution might empty")
```
Comment (craigcondit) [inline pkg/common/si_helper.go:45]: Let's use `log.ShimUtils` for this as that's logically where this goes (helper is a synonym for utility).

**Review by craigcondit — CHANGES_REQUESTED** [review]
Looking pretty good, some minor changes below.

**Comment by craigcondit** [comment]
@FrankYang0529 there are also a few merge conflicts caused by recent PRs that will need to be fixed, hopefully nothing too complex.

**Comment by FrankYang0529** [comment]
Hi @craigcondit, thanks for the review. I updated my PR.

**Review by craigcondit — APPROVED** [review]
+1. Looks good. Will merge once e2e tests finish.

**Pull request closed without merging**